test/parallel/test-assert.js

```diff
@@ -409,6 +409,10 @@ assert.throws(() => {
 }, /Got unwanted exception: user message/,
               'a.doesNotThrow ignores user message');
 
+assert.throws(() => {
+  assert.doesNotThrow(makeBlock(thrower, Error), 'user message');
+}, /Got unwanted exception: user message\ntest/);
```

Comment:
Also, probably a good idea to include a test case for if the primitive is thrown rather than a proper `Error`.